src/test/java/org/folio/rest/impl/OaiPmhImplTest.java

```diff
@@ -174,15 +184,49 @@ public void getOaiMetadataFormats(TestContext context) {
   }
 
   @Test
-  public void getOaiSets(TestContext context) {
+  public void testSuccessfulGetOaiSets(TestContext context) throws JAXBException {
     Async async = context.async();
 
     RequestSpecification requestSpecification = createBaseRequest();
-    String response = test500WithErrorMessage(requestSpecification, LIST_SETS_PATH);
+
+    String response = requestSpecification
+      .when()
+        .get(LIST_SETS_PATH)
+      .then()
+        .statusCode(200)
+        .contentType(ContentType.XML)
+        .extract()
+          .body().asString();
 
     // Check that error message is returned
     context.assertNotNull(response);
 
+    // Unmarshal string to OAIPMH and verify required data presents
+    OAIPMH oaipmh1FromString = ResponseHelper.getInstance().stringToOaiPmh(response);
+
+    verifyBaseResponse(oaipmh1FromString, LIST_SETS);
+    assertThat(oaipmh1FromString.getListSets(), is(notNullValue()));
```

Comment:
Not a big deal, but consider using the `context.assert*` methods for consistency... Especially since the order of expected/actual is reversed here.

Reply:
I would prefer to use hamcrest matchers instead of context.assert* within this module if there are no reasons to do otherwise. I see the rest of the team also tend to use it in their tests. 
So I replaced all context based asserts with hamcrest versions.